pycytominer/cyto_utils/cell_locations.py

```diff
@@ -0,0 +1,400 @@
+    - Each cell has at least 3 columns: `Nuclei_Location_Center_X`, `Nuclei_Location_Center_Y`, `ImageNumber`
+
+    In the `Image` table,
+    - Each row is an image
+    - Each image has at least the same columns as the images in the metadata file are indexed by, e.g., `Metadata_Plate`,`Metadata_Well`,`Metadata_Site`
+
+    The methods in this class do the following
+    - Read the metadata file
+    - Read the single_cell file
+    - For each image in the metadata file, find the corresponding image in the single_cell file
+    - For each cell in the corresponding image, find the X,Y location
+    - Add the X,Y locations of all cells to the metadata file in the corresponding row, packed into a single column
+
+
+    Attributes
+    ----------
+    metadata_input : str or Pandas DataFrame
+        Path to the input metadata file or a Pandas DataFrame
+
+    single_cell_input : str or sqlite3.Connection
+        Path to the single_cell file or a sqlite3.Connection object
+
+    augmented_metadata_output : str
+        Path to the output file. If None, the metadata file is not saved to disk
+
+    image_column : default = 'ImageNumber'
+        Name of the column in the metadata file that links to the single_cell file
+
+    object_column : default = 'ObjectNumber'
+        Name of the column in the single_cell file that identifies each cell
+
+    cell_x_loc : default = 'Nuclei_Location_Center_X'
+        Name of the column in the single_cell file that contains the X location of each cell
+
+    cell_y_loc : default = 'Nuclei_Location_Center_Y'
+        Name of the column in the single_cell file that contains the Y location of each cell
+
+    Methods
+    -------
+    add_cell_location()
+        Augment the metadata file and optionally save it to a file
+
+    """
+
+    def __init__(
+        self,
+        metadata_input: str or pd.DataFrame,
+        single_cell_input: str or sqlite3.Connection,
+        augmented_metadata_output: str = None,
+        overwrite: bool = False,
+        image_column: str = "ImageNumber",
+        object_column: str = "ObjectNumber",
+        image_index=["Metadata_Plate", "Metadata_Well", "Metadata_Site"],
+        cell_x_loc: str = "Nuclei_Location_Center_X",
+        cell_y_loc: str = "Nuclei_Location_Center_Y",
+    ):
+        self.metadata_input = self._expanduser(metadata_input)
+        self.augmented_metadata_output = self._expanduser(augmented_metadata_output)
+        self.single_cell_input = self._expanduser(single_cell_input)
+        self.overwrite = overwrite
+        self.image_column = image_column
+        self.object_column = object_column
+        self.image_index = image_index
+        self.cell_x_loc = cell_x_loc
+        self.cell_y_loc = cell_y_loc
+
+    def _expanduser(self, obj):
+        """Expand the user home directory in a path"""
+        if obj is not None and isinstance(obj, str) and not obj.startswith("s3://"):
+            return os.path.expanduser(obj)
```

Comment:
We've replaced `os` with `pathlib`, please update to pathlib (older functions have not yet been upgraded)

see https://docs.python.org/3/library/pathlib.html#pathlib.Path.expanduser

Reply:
Will do